sw/airborne/modules/guidance/gvf_parametric/gvf_parametric.cpp

```diff
@@ -345,14 +361,85 @@ bool gvf_parametric_2D_trefoil_wp(uint8_t wp, float w1, float w2, float ratio, f
 { 
   gvf_parametric_trajectory.p_parametric[7] = wp;
   gvf_parametric_plen_wps = 1;
-
+  #if defined(FIXEDWING_FIRMWARE)
   gvf_parametric_2D_trefoil_XY(waypoints[wp].x, waypoints[wp].y, w1, w2, ratio, r, alpha);
+  #elif defined(ROVER_FIRMWARE)
+  gvf_parametric_2D_trefoil_XY(WaypointX(wp), WaypointY(wp), w1, w2, ratio, r, alpha);
+  #endif
```

Comment:
The macros `WaypointX` and `WaypointY` exist for all firmwares, so it should work with `gvf_parametric_2D_trefoil_XY(WaypointX(wp), WaypointY(wp), w1, w2, ratio, r, alpha);` in all cases.